### PR TITLE
[CMake] FIX Debug build + more cleaning

### LINKED ISSUE #1851: runSofa_d.exe in Debug on Windows crash at start in CImg
Crash at
```
      // Allocate Memory for Image Read
      png_bytep *const imgData = new png_bytep[H];
      for (unsigned int row = 0; row<H; ++row) imgData[row] = new png_byte[(size_t)byte_depth*4*W];
      png_read_image(png_ptr,imgData);
      png_read_end(png_ptr,end_info);
```

I suspect the new background image...

### PR DESCRIPTION
**Pull request by guparan**
3 changes here:
- ~~FIX Debug build --> force all Zlib links to use the Release version~~
- CLEAN SOFA_HAVE_SOFA.YYY macro for consistency --> use SOFA_HAVE_SOFA_YYY everywhere
- FIX Boost::program_option copy --> `sofa_install_libraries` was missing

**IMPORTANT:** Windows users must update the `lib` directory in SOFA sources with https://www.sofa-framework.org/download/WinDepPack/latest/

Fixes #1851

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by jnbrunet** [comment]
Shouldn't be `SOFA_HAS_XXXX` ?

https://www.thesaurus.com/e/grammar/has-vs-have/

**Comment by guparan** [comment]
Historical reasons ... you know the drill :disappointed_relieved: 
Consistency is more important than grammatical accuracy :+1:

**Comment by fredroy** [comment]
seems that CMake uses HAVE_*
(see https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html or https://cmake.org/cmake/help/latest/module/CheckTypeSize.html ) 🧐

**Comment by guparan** [comment]
Back to WIP ... my fix was not a fix :+1: :+1: :+1:

**Comment by guparan** [comment]
Ok it's fixed now. Windows users must update the `lib` directory in SOFA sources with https://www.sofa-framework.org/download/WinDepPack/latest/ :+1:

File: applications/plugins/CImgPlugin/CMakeLists.txt [hunk 1]
```diff
@@ -88,6 +84,8 @@ sofa_create_package_with_targets(
     RELOCATABLE "plugins"
     )
 
+debug_print_target_properties(${PROJECT_NAME})
```
Comment (fredroy) [inline applications/plugins/CImgPlugin/CMakeLists.txt]: Intentional ?

Comment (guparan) [reply applications/plugins/CImgPlugin/CMakeLists.txt]: Nope :laughing:

**Pull request merged**